File: python/nominal2.py
from math import sqrt

class Rectangle:
    def __init__(
        self,
        width: int,
        height: int,
    ):
        self.width = width
        self.height = height

    def area(self) -> int:
        return self.width * self.height


class EquilateralTriangle:
    def __init__(self, side_len: int):
        self.side_len = side_len

    def area(self) -> float:
        return sqrt(3) / 4 * self.side_len ** 2


def round_area(shape: Rectangle) -> int:
    return int(round(shape.area()))


round_area(Rectangle(100, 50))
round_area(EquilateralTriangle(5))

File: typescript/nominal2.ts

class Rectangle {
  constructor(private width: number, private height: number) { }
  area = () => this.width * this.height
}

class EquilateralTriangle {
  constructor(private sideLen: number) { }
  area = () => Math.sqrt(3) / 4 * this.sideLen ** 2
}

export const roundArea = (shape: Rectangle) => Math.round(shape.area())

roundArea(new Rectangle(100, 50))
roundArea(new EquilateralTriangle(5))
